src/core/Interpreter.ts

```diff
@@ -168,25 +174,31 @@ class Interpreter {
         } else if (decodedData.txType === TxType.TRANSFER) {
             // Generic transfer
             interpretGenericTransfer(decodedData, interpretation)
-        } else if (interpretationMapping && methodSpecificMapping && methodName && toAddress) {
+        } else if (interpretationMapping && methodSpecificMappings.length && methodName && toAddress) {
             // Contract-specific interpretation
-
             // some of these will be arbitrary keys
             interpretation.contractName = interpretationMapping.contractName
-            interpretation.action = methodSpecificMapping.action // TODO: make a dynamic "action" field
-            interpretation.exampleDescription = methodSpecificMapping.exampleDescription
+
+            for (const methodSpecificMapping of methodSpecificMappings) {
+                if (methodSpecificMapping.action !== "__NFTSALE__") {
```

Comment:
I'd switch these to `if Action.__NFTSALE__` then `getActionFromInterpretation`

And then rename `getActionFromInterpretation` to something that clarifies this is the function we run specifically for `Action.__NFTSALE__`

Reply:
same feedback for when you use `getNativeValueTransferredFromInterpretation`, in terms of making it clear that function is specifically for `__NATIVEVALUETRANSFERRED__`